solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -944,6 +946,14 @@ public void load() {
       });
 
       clusterSingletons.setReady();
+      if (NodeRoles.PREFERRED.equals(nodeRoles.getRoleMode(NodeRoles.Role.OVERSEER))) {
+        try {
+          log.info("This node is started as a preferred overseer");
+          zkSys.getZkController().setPreferredOverseer();
+        } catch (KeeperException | InterruptedException e) {
+          throw new SolrException(ErrorCode.SERVER_ERROR, e);
+        }
+      }
       if (!distributedCollectionCommandRunner.isPresent()) {
```

Comment:
The added bloc above must be moved inside this if otherwise we will be sending a message over the overseer queue when no overseer is running to consume it.

Reply:
Couldn't understand the comment. @noblepaul please take a look.

Reply:
You can't do Overseer related actions if Overseer is not used in the cluster that is configured for distributed cluster state change and collection API.

So the code should instead look like:
```
if (!distributedCollectionCommandRunner.isPresent()) {
     if (NodeRoles.PREFERRED.equals(nodeRoles.getRoleMode(NodeRoles.Role.OVERSEER))) {
        try {
           log.info("This node is started as a preferred overseer");
           zkSys.getZkController().setPreferredOverseer();
        } catch (KeeperException | InterruptedException e) {
           throw new SolrException(ErrorCode.SERVER_ERROR, e);
        }
     }
     zkSys.getZkController().checkOverseerDesignate();
}
```

Reply:
Fixed, thanks.